File: examples/greeter.pl
#!/usr/bin/perl

# For each line FOO received on STDIN, respond with "Hello FOO!".

# Autoflush output
use IO::Handle;
STDOUT->autoflush(1);

# Read STDIN
while (<>) {
  chomp; # remove \n
  $visitors++;
  print "Hello $_!\n";
}

File: examples/ruby/greeter.rb
#!/usr/bin/ruby

# Copyright 2013 Joe Walnes and the websocketd team.
# All rights reserved.
# Use of this source code is governed by a BSD-style
# license that can be found in the LICENSE file.

# Autoflush output
STDOUT.sync = true

# For each line FOO received on STDIN, respond with "Hello FOO!".
while 1
  line = STDIN.readline.strip
  puts "Hello #{line}!"
end
